auth_token/models.py

```diff
@@ -129,3 +122,48 @@ def save(self):
 
     def delete(self):
         raise NotImplementedError
+
+
+class DeviceKeyQuerySet(models.QuerySet):
+
+    def create_and_get_token(self, uuid, user, user_agent=''):
+        """
+        This method must be called when user is authenticated.
+        It creates a new DeviceKey for the device and returns it.
+        If DeviceKey for the same UUID, device ID
+        and user already exists then it remain as it is and None is returned.
+        """
+        from auth_token.utils import generate_key
+        token = generate_key(length=64)
+        device_key, created = self.get_or_create(
+            uuid=uuid, is_active=True, user=user,
+            defaults={
+                'login_token': make_password(token),
+                'user_agent': user_agent[:256],
+            }
+        )
+        if created:
```

Comment:
proč se vrací jen když je created? A co se vrací v druhé větvi? Raději bych vracel obě device_key, created ať se ten kdo to využívá rozhodne cos tím

Reply:
ok, it returns (token, created) tuple now.